### ISSUE TITLE
Docs: Icon Only Buttons are not announced with screen readers 

### ISSUE THREAD

**Issue by coliff** [trimble-oss/modus-web-components#2077]
RE: https://moduswebcomponents.netlify.app/?path=/docs/components-button--icon-only#icon-only-button

Icon-only buttons need `aria-label` otherwise they will simply be announced as button which isn't helpful.

![image](https://github.com/trimble-oss/modus-web-components/assets/1212885/d040efce-14dc-45eb-a856-eac0f8e81813)

In this case the buttons here should have `aria-label="Notifications"`.

**Comment by rthanga1** [comment]
This is up to the user? We don't know which icon the user will use, right?
